fix-#392: added related blogs section in detailed blog page

vamsidhar-914 opened the pull request:
## Summary

added  related blogs section in detailed blog page

## Description

whenever user visited individual blog page..they can see related posts based on the categories in the blog page

## Images
desktop view
![Screenshot (200)](https://github.com/krishnaacharyaa/wanderlust/assets/127492265/b8232dc8-0838-4168-b57b-c999c717d834)

mobile view
![Screenshot (201)](https://github.com/krishnaacharyaa/wanderlust/assets/127492265/cb08e2d9-fd62-414b-a470-fb3f30b6c071)


## Issue(s) Addressed

Closes #392 

## Prerequisites

- [ yes] Have you followed all the [CONTRIBUTING GUIDELINES](https://github.com/krishnaacharyaa/wanderlust/blob/main/.github/CONTRIBUTING.md#guidelines-for-contributions)?

vamsidhar-914 commented:
hey @krishnaacharyaa please do review it and want any changes then tell me 👍

vamsidhar-914 commented:
helllo @krishnaacharyaa ??

krishnaacharyaa commented on frontend/src/pages/details-page.tsx:
```diff
@@ -30,6 +37,24 @@ export default function DetailsPage() {
   }, [post]);
 
+  const category = post.categories[0];
+  useEffect(() => {
```
It should use all the categories, 
Basically if there are three categories a b and c , i need to get all the posts which has any of a b or c. and then get only top 3 posts in the mobile view, vertically and may be 4 or 5 in the desktop view

krishnaacharyaa commented on frontend/src/pages/details-page.tsx:
```diff
@@ -12,6 +17,8 @@ export default function DetailsPage() {
   const navigate = useNavigate();
+  const [categoryPosts, setcategoryPosts] = useState<Post[]>([]);
+  const [postLoading, setpostLoading] = useState(false);
 
```
I see this is declared, but never used anywhere, ideally this should be used and the naming should be relatedPostsLoading and shimmer view needs to be provided when it is loading

krishnaacharyaa reviewed (changes requested):
Thank you for the PR @vamsidhar-914 kindly address the review comments

vamsidhar-914 commented:
ok sure @krishnaacharyaa on itt

vamsidhar-914 commented:
hi @krishnaacharyaa 
 
fetched all related category posts and displayed 3 posts in mobile view and 4 posts in desktop view

krishnaacharyaa reviewed (changes requested):
I really appreciate the efforts, but we have a little of tasks to be done, kindly address the review comments

krishnaacharyaa commented:
For mobile can we aim for this kind (style) of related posts? 
<img width="1055" alt="image" src="https://github.com/krishnaacharyaa/wanderlust/assets/116620586/e26ce320-da54-4622-9da4-185dfd19639e">

small icon in the left and then the content on the right.

If you feel these are quite a few requests in a single PR, kindly let me know we can break this into subtasks, and we can raise this as another PR :) I'm totally fine with that

vamsidhar-914 commented:
no its finee...i'll be on it

vamsidhar-914 commented:
@krishnaacharyaa 

changed naming conventions
added mobile responsive for related posts
![Screenshot (206)](https://github.com/krishnaacharyaa/wanderlust/assets/127492265/a4108699-f4bb-4fc0-92ba-08aa32ea9b4f)

vamsidhar-914 commented:
@krishnaacharyaa  ?

krishnaacharyaa commented:
Super @vamsidhar-914 
Thank you for the efforts, really appreciated.

Can you use 
<svg xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink" width="512" height="512" x="0" y="0" viewBox="0 0 492.004 492.004" style="enable-background:new 0 0 512 512" xml:space="preserve"><g><path d="M382.678 226.804 163.73 7.86C158.666 2.792 151.906 0 144.698 0s-13.968 2.792-19.032 7.86l-16.124 16.12c-10.492 10.504-10.492 27.576 0 38.064L293.398 245.9l-184.06 184.06c-5.064 5.068-7.86 11.824-7.86 19.028 0 7.212 2.796 13.968 7.86 19.04l16.124 16.116c5.068 5.068 11.824 7.86 19.032 7.86s13.968-2.792 19.032-7.86L382.678 265c5.076-5.084 7.864-11.872 7.848-19.088.016-7.244-2.772-14.028-7.848-19.108z" fill="#000000" opacity="1" data-original="#000000"></path></g></svg>

For the arrow right svg
May be this is black version, make white version of it by replacing #000000 with #ffffff and make 2 svgs and kindly handle both dark and light theme


And
<img width="255" alt="image" src="https://github.com/krishnaacharyaa/wanderlust/assets/116620586/9b3ca341-9e0a-4cfe-b64f-38e02b8c3190">

Can you please take care of this alignment, here the see more related blogs > should be aligned to the below card like 
<img width="617" alt="image" src="https://github.com/krishnaacharyaa/wanderlust/assets/116620586/4958c137-488b-48b9-b599-d3e37ccf8d94">

And kindly upload the photo of the shimmer/skeleton things aswell, if you feel its too much in a single PR we can take this shimmer loading skeletons in another PR i am fine with that :)

IF you could close the above 2 bugs we are good to merge this PR :)

vamsidhar-914 commented:
fixed svg icons for light and dark mode

added loading skeletons
![Screenshot (208)](https://github.com/krishnaacharyaa/wanderlust/assets/127492265/8437e963-30a0-42c5-bfd2-7fd4e54ec4af)

is this alignment ok? or u want any change to it

krishnaacharyaa commented:
> @krishnaacharyaa
> 
> fixed svg icons for light and dark mode
> 
> added loading skeletons ![Screenshot (208)](https://private-user-images.githubusercontent.com/127492265/337439325-8437e963-30a0-42c5-bfd2-7fd4e54ec4af.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc3MjYzMDUsIm5iZiI6MTcxNzcyNjAwNSwicGF0aCI6Ii8xMjc0OTIyNjUvMzM3NDM5MzI1LTg0MzdlOTYzLTMwYTAtNDJjNS1iZmQyLTdmZDRlNTRlYzRhZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYwN1QwMjA2NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02YTRmNzRmZjY4Yjg3MjQ1NWE4ZTg4MGVkY2E4ZmRhZjE5OWI5MDYwYjVlNTdmY2E0YTE1M2NmZTg0YzM3MzFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.pmRNI5iSYy7WgfC9kG-2J6G2lBZ9MpKTwvK3sJYiZ1s)
> 
> is this alignment ok? or u want any change to it

Can't we have it right side as earlier 
Just the end should match

You need to use something like justify-between

krishnaacharyaa commented:
Here I'm not talking about the skeletons I'm just talking about the "Related blogs" and "See all related blogs" that can we have in a single row! That would be nice
Just it should be end aligned as I marked in red line in previous comment

krishnaacharyaa commented on frontend/src/components/PostMobileViewCardSkeleton.tsx:
I guess this should go under skeletons folder, not sure, whereever all the skeletons are placed, kindly place it there,
And naming sir, please follow guidelines `related-post-card-skeleton` as other skeletons namings.

vamsidhar-914 commented:
done it sir @krishnaacharyaa !

krishnaacharyaa commented:
Thank you sir :)

krishnaacharyaa reviewed (approved):
LGTM, Thank you for all the constant interest and enthusiasm and completing this task :)
Really appreciated
